silkworm/node/execution/block/block_executor.cpp

```diff
@@ -0,0 +1,117 @@
+/*
+   Copyright 2024 The Silkworm Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+#include <tuple>
+
+#include <silkworm/core/execution/call_tracer.hpp>
+#include <silkworm/core/execution/processor.hpp>
+#include <silkworm/infra/common/log.hpp>
+#include <silkworm/node/execution/block/block_executor.hpp>
+
+namespace silkworm::execution::block {
+
+using namespace std::chrono_literals;
+
+//! Generate log arguments for execution progress at specified block
+static log::Args log_args_for_exec_progress(BlockExecutor::ExecutionProgress& progress, uint64_t current_block) {
```

Comment:
This logging format is specifically crafted for Erigon++ to match Erigon logs and must be kept into the C API implementation. The stage pipeline has its own timer-based logging mechanism that must remain the same because it's shared among all the stages.